### PR TITLE
Add checkov (fixes #39)

### LINKED ISSUE #39: Add checkov for terraform
Depends on #37 

We should add `checkov` as a security tool prior to executing `terraform` commands

### PR DESCRIPTION
**Pull request by JonZeolla**
# Contributor Comments
Adds `checkov` in addition to `tfsec` for `terraform`.  Depends on #42

## Pull Request Checklist
Thank you for submitting a contribution to `easy_infra`!

In order to streamline the review of your contribution we ask that you review and comply with the below requirements:
 - [X] Ensure that `make build` completes successfully
 - [X] Rebase your branch against the latest commit of the target branch, which likely should be `master`
 - [X] If there is an issue associated with your Pull Request, align your PR title with [this documentation](https://help.github.com/en/articles/closing-issues-using-keywords)

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -68,10 +68,11 @@ RUN git clone https://github.com/tfutils/tfenv.git ~/.tfenv \
  && tfenv use ${TERRAFORM_VERSION}
 
 # pip installs
-COPY awscli.txt .
+COPY awscli.txt checkov.txt .
 ENV PATH="/root/.local/bin:${PATH}"
 RUN python3 -m pip install --upgrade --no-cache-dir pip \
- && pip install --user --no-cache-dir -r awscli.txt
+ && pip install --user --no-cache-dir -r awscli.txt \
```
Comment (JonZeolla) [inline Dockerfile:74]: I would have just done `-r awscli.txt -r checkov.txt` but they depend on conflicting versions of `boto`

File: Makefile [hunk 2]
```diff
@@ -30,20 +30,20 @@ update_dockerfile_repo    = ./update_components.sh --repo=$(1) --version=$(2)
 all: update build
 
 .PHONY: update
-update: update-dependencies update-functions
+update: update-dependencies
 
 .PHONY: update-dependencies
-update-dependencies: update-apt update-requirements update-awscli update-github update-terraform
+update-dependencies: update-apt update-ci update-awscli update-checkov update-github update-terraform
```
Comment (JonZeolla) [inline Makefile:36]: I expect that in the future we will want to refactor this to be more dynamic.

**Comment by JonZeolla** [comment]
@pb-seiso The dependencies are merged and I've rebased onto `master` so this is ready for review.

File: easy_infra.yml [hunk 3]
```diff
@@ -10,3 +10,6 @@ commands:
       tfsec:
         command: tfsec .
         description: recursive scan
+      checkov:
+        command: checkov -d . --no-guide
```
Comment (pb-seiso) [inline easy_infra.yml:14]: I'm not sure that this works, assuming "checkov" should alias to this. Test command:

`docker run -v (pwd):/iac easy_infra /bin/bash -c "checkov"`

Result:
```

       _               _              
   ___| |__   ___  ___| | _______   __
  / __| '_ \ / _ \/ __| |/ / _ \ \ / /
 | (__| | | |  __/ (__|   < (_) \ V / 
  \___|_| |_|\___|\___|_|\_\___/ \_/  
                                      
By bridgecrew.io | version: 1.0.667 
Visualize and collaborate on security issues with Bridgecrew! 
Bridgecrew's dashboard allows automation of future checks, Pull Request scanning and auto-comments, automatic remidiation PR's and more! 
 Plus it's free for 100 cloud resources and a great way to visualize and collaborate on Checkov results. For more information on dashboard integration, see: http://bridge.dev/checkov-dashboard 
 
 To instantly see future Checkov scans in the platform, Press y! 

Visualize results? (y/n): Traceback (most recent call last):
  File "/root/.local/bin/checkov", line 5, in <module>
    run()
  File "/root/.local/lib/python3.8/site-packages/checkov/main.py", line 90, in run
    bc_integration.onboarding()
  File "/root/.local/lib/python3.8/site-packages/checkov/common/bridgecrew/platform_integration.py", line 220, in onboarding
    reply = self._input_visualize_results()
  File "/root/.local/lib/python3.8/site-packages/checkov/common/bridgecrew/platform_integration.py", line 322, in _input_visualize_results
    result = str(input('Visualize results? (y/n): ')).lower().strip()  # nosec
EOFError: EOF when reading a line
```

When running interactively and calling `checkov -d . --no-guide`, the result is:
```
ERROR:checkov.terraform.checks.resource.aws.SecurityGroupUnrestrictedIngress22:Failed to run check: Ensure no security groups allow ingress from 0.0.0.0:0 to port 22 for configuration: {'type': ['ingress'], 'cidr_blocks': [['0.0.0.0/0']]} at file: /main.tf
2020-12-11 16:42:26,765 [MainThread  ] [ERROR]  Failed to run check: Ensure no security groups allow ingress from 0.0.0.0:0 to port 22 for configuration: {'type': ['ingress'], 'cidr_blocks': [['0.0.0.0/0']]} at file: /main.tf
Traceback (most recent call last):
  File "/root/.local/bin/checkov", line 5, in <module>
    run()
  File "/root/.local/lib/python3.8/site-packages/checkov/main.py", line 69, in run
    scan_reports = runner_registry.run(root_folder=root_folder, external_checks_dir=external_checks_dir,
  File "/root/.local/lib/python3.8/site-packages/checkov/common/runners/runner_registry.py", line 32, in run
    scan_report = runner.run(root_folder, external_checks_dir=external_checks_dir, files=files,
  File "/root/.local/lib/python3.8/site-packages/checkov/terraform/runner.py", line 64, in run
    self.check_tf_definition(report, root_folder, runner_filter, collect_skip_comments)
  File "/root/.local/lib/python3.8/site-packages/checkov/terraform/runner.py", line 92, in check_tf_definition
    self.run_all_blocks(definition, definitions_context, full_file_path, root_folder, report,
  File "/root/.local/lib/python3.8/site-packages/checkov/terraform/runner.py", line 99, in run_all_blocks
    self.run_block(definition[block_type], definitions_context,
  File "/root/.local/lib/python3.8/site-packages/checkov/terraform/runner.py", line 139, in run_block
    results = registry.scan(scanned_file, entity, skipped_checks, runner_filter)
  File "/root/.local/lib/python3.8/site-packages/checkov/common/checks/base_check_registry.py", line 103, in scan
    result = self.run_check(check, entity_configuration, entity_name, entity_type, scanned_file, skip_info)
  File "/root/.local/lib/python3.8/site-packages/checkov/common/checks/base_check_registry.py", line 109, in run_check
    result = check.run(scanned_file=scanned_file, entity_configuration=entity_configuration,
  File "/root/.local/lib/python3.8/site-packages/checkov/common/checks/base_check.py", line 58, in run
    raise e
  File "/root/.local/lib/python3.8/site-packages/checkov/common/checks/base_check.py", line 39, in run
    check_result['result'] = self.scan_entity_conf(entity_configuration, entity_type)
  File "/root/.local/lib/python3.8/site-packages/checkov/terraform/checks/resource/base_resource_check.py", line 16, in scan_entity_conf
    return self.scan_resource_conf(conf, entity_type)
  File "/root/.local/lib/python3.8/site-packages/checkov/terraform/checks/resource/base_resource_check.py", line 28, in wrapper
    return wrapped(self, conf)
  File "/root/.local/lib/python3.8/site-packages/checkov/terraform/checks/resource/aws/AbsSecurityGroupUnrestrictedIngress.py", line 50, in scan_resource_conf
    return CheckResult.FAILED if self.contains_violation(conf) else CheckResult.PASSED
  File "/root/.local/lib/python3.8/site-packages/checkov/terraform/checks/resource/aws/AbsSecurityGroupUnrestrictedIngress.py", line 58, in contains_violation
    from_port = force_int(force_list(conf['from_port'])[0])
KeyError: 'from_port'
```

Is this expected behavior?

Comment (JonZeolla) [reply easy_infra.yml:14]: Yes, it's why I added the `terraform init && terraform validate` steps in #42

**Comment by JonZeolla** [comment]
I agree; that is ripe for a refactor.

**Review by pb-seiso — DISMISSED** [review]

**Review by pb-seiso — APPROVED** [review]

**Review by pb-seiso — APPROVED** [review]

**Pull request merged**